exercises/go-counting/canonical-data.json

```diff
@@ -0,0 +1,172 @@
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Test for black territory on 5x5 board",
+      "property": "territory",
+      "input": {
+        "board": [
+                    "  B  ",
+                    " B B ",
+                    "B W B",
+                    " W W ",
+                    "  W  "
+                 ],
+        "x": 0,
+        "y": 1
+      },
+      "expected": {
+        "stone": "BLACK",
+        "territory": [[0, 0], [0, 1], [1, 0]]
+      }
+    },
+    {
+      "description": "Test for white territory on 5x5 board",
+      "property": "territory",
+      "input": {
+        "board": [
+                    "  B  ",
+                    " B B ",
+                    "B W B",
+                    " W W ",
+                    "  W  "
+                ],
+        "x": 2,
+        "y": 3
+      },
+      "expected": {
+        "stone": "WHITE",
+        "territory": [[2, 3]]
+      }
+    },
+    {
+      "description": "Test for open territory on 5x5 board",
+      "property": "territory",
+      "input": {
+        "board": [
+                    "  B  ",
+                    " B B ",
+                    "B W B",
+                    " W W ",
+                    "  W  "
+                ],
+        "x": 1,
+        "y": 4
+      },
+      "expected": {
+        "stone": "NONE",
+        "territory": [[0, 3], [0, 4], [1, 4]]
+      }
+    },
+    {
+      "description": "Test for none territory on 5x5 board",
+      "property": "territory",
+      "input": {
+        "board": [
+                    "  B  ",
+                    " B B ",
+                    "B W B",
+                    " W W ",
+                    "  W  "
+                ],
+        "x": 1,
+        "y": 1
+      },
+      "expected": {
+        "stone": "NONE",
+        "territory": []
+      }
+    },
+    {
+      "description": "Test for valid coordinate on 5x5 board",
+      "property": "territory",
+      "input": {
+        "board": [
+                    "  B  ",
+                    " B B ",
+                    "B W B",
+                    " W W ",
+                    "  W  "
+                ],
+        "x": -1,
+        "y": 1
+      },
+      "expected": {
+        "stone": "NONE",
+        "territory": []
+      }
+    },
+    {
+      "description": "Test for valid coordinate #2 on 5x5 board",
+      "property": "territory",
+      "input": {
+        "board": [
+                    "  B  ",
+                    " B B ",
+                    "B W B",
+                    " W W ",
+                    "  W  "
+                ],
+        "x": 1,
+        "y": 5
+      },
+      "expected": {
+        "stone": "NONE",
+        "territory": []
+      }
+    },
+    {
+      "description": "Test when one territory is the whole board",
+      "property": "territories",
+      "input": {
+        "board": [
+                    " "
+                ]
+      },
+      "expected": {
+        "territoryBlack": [],
```

Comment:
I notice this says `territory`, singular. What if a board has more than one black territory? How should that be handled?

Reply:
Well it would still all be considered black's territory, but I'll handle it however you want me to. I can change it to territoriesBlack if you like.

Reply:
> it would still all be considered black's territory

true story - then the question is whether they should all go in the same array. For the following board:

```json
"board": [
  " B "
]
```

Should the answer for black be a single array `[[0, 0], [0, 2]]`, or should it be two, one for each contiguous region?

```json
[
 [
    [0, 0]
 ],
 [
    [0, 2]
 ]
]
```

I think I had misunderstood the concept of territory. I thought that each contiguous region is a single territory. This understanding stems from the fact that `territory` designates a single intersection on the board and only examines the single region that contains that intersection. But it could be instead that each player has a single territory that may comprise multiple regions.

If I didn't misunderstand, it would want to be `territories` for each player (and each contiguous region would be in its own array), OR it would just be an array whose elements are the same type as the `expected` value of each `territory`.

If someone would confirm to me that I misunderstood, then it would make sense to keep it as `territory`.

Reply:
EBWODP: Previous comment did not sufficiently make clear the distinction between `territory`/`territories` the properties being tested versus the abstract concept in the Go game. I hope this edit makes it clearer:

I think I had misunderstood the concept of territory. I thought that each contiguous region is a single territory. This understanding stems from the fact that `territory` (the property being tested earlier in this file) designates a single intersection on the board and only examines the single region that contains that intersection. But it could be instead that each player has a single territory that may comprise multiple regions.

If I didn't misunderstand, then the keys in the expected object of `territories` would each want to be named `territories` for each player (and each contiguous region would be in its own array), OR the `expected` object of a `territories` case would just be an array whose elements are the same type as the `expected` value of a `territory` case.

If someone would confirm to me that I misunderstood, then for the expected answer to `territories`, it would make sense to keep the key names as `territory` for each player .